scylla/src/history.rs

```diff
@@ -58,3 +61,145 @@ pub trait HistoryListener: Debug + Send + Sync {
         retry_decision: &RetryDecision,
     );
 }
+
+pub type TimePoint = DateTime<Utc>;
+
+/// HistoryCollector can be used as HistoryListener to collect all the query history events.
+/// Each event is marked with an UTC timestamp.
+#[derive(Debug, Default)]
+pub struct HistoryCollector {
+    data: Mutex<HistoryCollectorData>,
```

Comment:
I assumed that we don't care about efficiency at all.
This will most likely be used in debugging contexts.

The interface defined by `HistoryListener` trait is generic enough that we can implement a more efficient collector if the need arises.